sw/device/silicon_creator/rom/e2e/BUILD

```diff
@@ -182,12 +193,89 @@ opentitan_functest(
         timeout = "eternal",
         rom = "//sw/device/silicon_creator/rom",
     ),
-    deps = [
-        "//sw/device/lib/dif:keymgr",
-        "//sw/device/lib/testing:keymgr_testutils",
-        "//sw/device/lib/testing:otp_ctrl_testutils",
-        "//sw/device/lib/testing/test_framework:ottf_main",
-    ],
+)
+
+rom_e2e_keymgr_init_configs = [
+    {
+        "name": "rom_ext_meas",
+        "value": CONST.TRUE,
+    },
+    {
+        "name": "rom_ext_no_meas",
+        "value": CONST.FALSE,
+    },
+    {
+        "name": "rom_ext_invalid_meas",
+        "value": 0,
+    },
+]
+
+[
+    otp_json(
+        name = "otp_json_keymgr_{}".format(config["name"]),
+        partitions = [
+            otp_partition(
+                name = "OWNER_SW_CFG",
+                items = {
+                    "OWNER_SW_CFG_ROM_KEYMGR_ROM_EXT_MEAS_EN": hex(config["value"]),
+                },
+            ),
+        ],
+    )
+    for config in rom_e2e_keymgr_init_configs
+]
+
+[
+    otp_image(
+        name = "otp_img_keymgr_{}".format(config["name"]),
+        src = "//hw/ip/otp_ctrl/data:otp_json_rma",
+        overlays = STD_OTP_OVERLAYS + [":otp_json_keymgr_{}".format(config["name"])],
+        visibility = ["//visibility:private"],
+    )
+    for config in rom_e2e_keymgr_init_configs
+]
+
+[
+    bitstream_splice(
+        name = "bitstream_keymgr_{}".format(config["name"]),
+        src = "//hw/bitstream:rom",
+        data = ":otp_img_keymgr_{}".format(config["name"]),
+        meminfo = "//hw/bitstream:otp_mmi",
+        tags = ["vivado"],
+        visibility = ["//visibility:private"],
+    )
+    for config in rom_e2e_keymgr_init_configs
+]
+
+[
+    opentitan_functest(
+        name = "rom_e2e_keymgr_init_{}".format(config["name"]),
+        cw310 = cw310_params(
+            bitstream = ":bitstream_keymgr_{}".format(config["name"]),
+            tags = ["vivado"],
+        ),
+        dv = dv_params(
+            rom = "//sw/device/silicon_creator/rom",
+        ),
+        ot_flash_binary = ":rom_e2e_keymgr_init_test",
```

Comment:
because are setting the flash image like this, in DV, you need to add the `:ot_flash_binary` tag to the test description [here](https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/dv/chip_rom_tests.hjson;drc=84aaf54d100214bd87212ff697c2f2d5c9013e14;l=309). Also you will need to:
1. update the dv SW image name to match
2. add a separate entry for the OTP SW image name in the `sw_images` list (see [here](https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/dv/chip_rom_tests.hjson;drc=84aaf54d100214bd87212ff697c2f2d5c9013e14;l=254) as an example)
3. add `"+use_otp_image=OtpTypeCustom" to the run_opts
4. duplicate the DV test [entry](https://cs.opensource.google/opentitan/opentitan/+/master:hw/top_earlgrey/dv/chip_rom_tests.hjson;drc=84aaf54d100214bd87212ff697c2f2d5c9013e14;l=306) for each LC state
5. and lastly test it all with DVsim to make sure we don't break the regression.

Lmk if you need help with any of that ^

Reply:
re: breaking the regression, I think that will happen on line 186, and not here.
I'm tempted to fully revert the original functest to avoid these issues.

Reply:
I was asked to prioritize the FPGA work over the DV work, so I think this is a PR that enables it on the FPGA and wont break DV.

I'll file an issue to followup with the above tasks

Reply:
no worries, thanks for clarifying and filing the issue @drewmacrae !

Reply:
#16696 contains the fixes / cleanup for DV